### PR TITLE
nvme.get_nsid

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by yocalebo** [comment]
A minor issue, but the copyright year should be set to 2019 :smiley:

File: nvme.pyx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+cdef extern from "dev/nvme/nvme.h":
+    int NVME_GET_NSID
+
+    struct nvme_get_nsid:
+        char cdev[256]
+        uint32_t nsid
+
+
+def get_nsid(path):
+    fd = os.open(path, os.O_RDWR)
```
Comment (william-gr) [inline nvme.pyx]: Does it need to be open as RDWR?
I wonder if OS will let you if there is a WR exclusive lock to that device?

File: nvme.pyx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    cdef nvme_get_nsid nsid
+    res = ioctl(fd, NVME_GET_NSID, &nsid)
```
Comment (william-gr) [inline nvme.pyx]: We should probably release GIL for the ioctl, just to be safe?

**Review by william-gr — APPROVED** [review]

**Comment by william-gr** [comment]
@amotin care to review this too, please?

**Review by amotin — APPROVED** [review]
I have no objections, just a comment that the function get_nsid() here does not actually return namespace ID, but only a controller path.  While devices presently on market out of the box support/have only one namespace 0, some allow multiple to be configured.

**Pull request merged**